acme/build.gradle

```diff
@@ -1,44 +1,18 @@
 plugins {
-    id "io.micronaut.build.internal.module"
-}
-
-configurations {
-    documentation
+    id 'io.micronaut.build.internal.acme-module'
 }
 
 dependencies {
-    annotationProcessor("io.micronaut:micronaut-inject-java")
-    annotationProcessor "io.micronaut.docs:micronaut-docs-asciidoc-config-props:$micronautDocsVersion"
-
-    implementation("io.micronaut:micronaut-http")
-    implementation("io.micronaut:micronaut-http-server")
-    implementation("io.micronaut:micronaut-http-server-netty")
-    implementation("io.micronaut:micronaut-inject")
-    implementation 'jakarta.inject:jakarta.inject-api:2.0.1'
+    implementation libs.micronaut.runtime
+    implementation libs.micronaut.http.server.netty
 
-    implementation "org.shredzone.acme4j:acme4j-client:${acmeVersion}"
-    implementation "org.shredzone.acme4j:acme4j-utils:${acmeVersion}"
+    api libs.managed.acme4j.client
+    api libs.managed.acme4j.utils
```

Comment:
Not sure these should be `api` dependencies, nor managed.  They were implementation before 🤔

Reply:
@melix please correct me if I'm wrong, all dependencies which must be exported by the library should be with `api` configuration. I have added the `managed` prefix because this dependency should be included inside the `acme-bom` module.

Reply:
My issue was that previously they were 

```
implementation "org.shredzone.acme4j:acme4j-client:${acmeVersion}"
```

So not exported, but now they are.  I am not sure if this was a previous oversight or a breaking change 🤔

Reply:
From my point of view, these two acme4j dependencies must be exported because all code inside this module requires them and will not work without them. As I remember sometime `implementation` dependencies can be exported as well, that's why it was worked correctly even before. But `implementation` doesn't guarantee it will export dependencies. I don't see this as a breaking change. Here is an explanation for this difference:

https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation

Reply:
@timyates I can restore to `implementation` back if you think it's better to do.

Reply:
Will users need to use reference any of the classes in `org.shredzone.acme4j:acme4j-client` in a Micronaut application using the Micronaut ACME module?

If yes, I think we can change them to api, if not lets leave it as implementation to avoid exposing the implementation detail.

Reply:
> From my point of view, these two acme4j dependencies must be exported because all code inside this module requires them and will not work without them.

Both `api` and `implementation` dependencies are exported, transitively. The difference is therefore not if they are exported or not, but to _what_ they are exported. It depends on the consumer. `api` dependencies are exported for _compilation of the consumer_. `implementation` dependencies are exported for _runtime_ of the consumer.

Therefore, you should use managed dependencies for `api` (it's possible to use managed dependencies for `implementation` too). To tell if you should use `api` or `implementation`, you should look at whether the types from the dependency are used in the ABI of the component or not (signatures of methods, superclasses, ...).